text/0000-interactive-app-creation.md

```diff
@@ -0,0 +1,142 @@
+1. How many questions do we think developers will tolerate?
+1. Should the questions be different if it is an app vs addon?
+
+We will also endeavor to provide strong defaults for these questions, based on the results from the community surveys we have been conducting over the past few years and the defaults as stated by Ember itself (either explicitly or implicitly). **If these defaults change over time as the result of community RFCs, the default values or other responses for the interactive questions should be updated as part of those RFCs.**
+
+So, how then do we decide which questions to include and which not to include? It became necessary to then develop a rubric by which the proposed list of questions could be measured. 
+
+### Question Rubric
+We intend for this section to be normative; that is, we intend for it to define how questions are evaluated for inclusion both now and in the future. 
+
+Evaluating criteria: 
+1. Is it hard (or time-consuming) to change it later? *For example, it's easy to remove the `ember-welcome-page` addon, so we did not include it. But it's time consuming to change from `yarn` to `npm` (or vice-versa) so we made sure to include that.* 
+1. Is it difficult to discover the option?
+1. What is the maintenance cost? *For example, a nice question to add could be something like, Do you want to use a CSS preprocessor? But, because Ember does not deal with SASS, LESS... by default, the interactive workflow will have to install an addon (like `ember-cli-less`) which would imply that the CLI team will have to ensure it never fails.* 
+1. Does the question affect another one? *For example, the LTS question can affect the CI file generated by the CI question (setup tests on CI with `ember-try`). Questions that modify the result of other questions too much should be avoided to prevent high maintenance.*
+
+### Triggering the Workflow
+
+The following commands, and reasonable combinations thereof, should work to enter into the interactive workflow: 
+
+1. `ember new`
+1. `ember new --interactive`
+1. `ember new -i`
+1. `ember new my-app -i`
+
+### App/Addon 
+
+Starting a new Ember app or addon as we do now (at the time of this writing) will not change. If a user types `ember new my-app` they will still get a new Ember app generated in the usual ways. However, if the user types `ember new` they will enter into an interactive workflow that will ask them a few key questions about their application. 
+
+If a user were to type in existing flags that are also questions in the interactive workflow (i.e., `ember new my-app --lang en-US --interactive`), it will skip the question entirely. 
+
+When a user types `ember new` into their command line, they will be asked these questions (defaults indicated by a `>`):
+
+- `Is this an app or an addon? (use arrow keys)`
+  - `> app`
+  - `addon`
+- `Please provide the name of your app/addon:`
+- `Please provide the spoken/content language of your app/addon: (use arrow keys)`
```

Comment:
Suggestion:

What is the default language of your app/addon? Please type your ISO639-1 language code. For example, the language code for English is "en" and the language code for Chinese is "chi". A full list can be found at https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes .

Motivation:

* Ensure equality for all languages
* Avoids discussion about which languages are "common"
* The wording for "default" gives people who are planning to do a multilingual app a path forward, whereas, the text above doesn't.

Reply:
For no real reason I've always used `en-US` for the language codes, never simple `en`. Would this make any difference?

Reply:
I can imagine that this would for example determine the accent used by a screenreader if no default is set and different accents (e.g. British, US, Australian) are installed. But I would have to verify to be sure.

Reply:
> For no real reason I've always used `en-US` for the language codes, never simple `en`. Would this make any difference?

@MichalBryxi it can change things in small ways, like spellcheck, how things are pronounced, etc.